Riot/Modules/Room/CellData/RoomBubbleCellData.m

```diff
@@ -1080,45 +1070,47 @@ - (NSString*)accessibilityLabelForAttachmentType:(MXKAttachmentType)attachmentTy
 
 #pragma mark - URL Previews
 
-- (void)loadURLPreview
+- (void)refreshURLPreviewForEventId:(NSString *)eventId
 {
-    // Get the last bubble component as that contains the link.
-    MXKRoomBubbleComponent *lastComponent = bubbleComponents.lastObject;
-    if (!lastComponent || !lastComponent.link)
+    // Get the event's component, but only if it has a link.
+    MXKRoomBubbleComponent *component = [self bubbleComponentWithLinkForEventId:eventId];
+    if (!component)
     {
         return;
     }
     
     // Don't show the preview if it has been dismissed already.
-    self.showURLPreview = ![URLPreviewService.shared hasClosedPreviewFrom:lastComponent.event];
-    if (!self.showURLPreview)
+    component.showURLPreview = ![URLPreviewService.shared hasClosedPreviewFrom:component.event];
```

Comment:
I'm wondering if we should set the `component.showURLPreview` to false when the `RiotSettings.shared.roomScreenShowsURLPreviews == false` in order to centralise the `RiotSettings.shared.roomScreenShowsURLPreviews` check.

Reply:
Ah, yes I like this idea 🙏